### PR TITLE
Check consistency of SVRL reports with notice examples

### PR DESCRIPTION
**Pull request by bertrand-lorentz**
Check that there is a one-to-one correspondance between notice examples and SVRL reports (TEDEFO-1799).
Check that notice examples that are supposed to be valid do not have errors in their SVRL reports (TEDEFO-1800).

Add corresponding unit tests.

### CODE REVIEW

File: src/main/resources/eu/europa/ted/eforms/sdk/analysis/drools/xmlNoticeRules.drl [hunk 1]
```diff
@@ -9,4 +9,29 @@ when
...
 then
   results.add(new ValidationResult($n, "the value in CustomizationID is incorrect", ValidationStatusEnum.ERROR));
-end
\ No newline at end of file
+end
+
+rule "Every notice example has a corresponding validation report"
+when
+  /xmlNotices[ $n: this, $exampleId : id ]
+  and
```
Comment (meletev) [inline src/main/resources/eu/europa/ted/eforms/sdk/analysis/drools/xmlNoticeRules.drl:17]: Do these "and" operators actually do anything?

File: src/main/java/eu/europa/ted/eforms/sdk/analysis/fact/SvrlReportFact.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package eu.europa.ted.eforms.sdk.analysis.fact;
+
+import eu.europa.ted.eforms.sdk.domain.SvrlReport;
+
+public class SvrlReportFact implements SdkComponentFact<String> {
+
+	private final SvrlReport svrlReport;
+
+	public SvrlReportFact(SvrlReport svrlReport) {
+		this.svrlReport = svrlReport;
```
[the remark this answers is not in the document]
Comment (bertrand-lorentz) [reply src/main/java/eu/europa/ted/eforms/sdk/analysis/fact/SvrlReportFact.java:13]: Added null check

**Review by rouschr — APPROVED** [review]

File: src/main/java/eu/europa/ted/eforms/sdk/analysis/fact/SvrlReportFact.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bertrand-lorentz) [reply src/main/java/eu/europa/ted/eforms/sdk/analysis/fact/SvrlReportFact.java]: My editor goes back to indenting with tabs on new files. I thought I had caught them all, but missed this one.
Fixed in new commit.

File: src/main/resources/eu/europa/ted/eforms/sdk/analysis/drools/xmlNoticeRules.drl [hunk 3]
```diff
@@ -9,4 +9,29 @@ when
...
+	not (exists /svrlReports[ id == $exampleId ])
```
[the remark this answers is not in the document]
Comment (bertrand-lorentz) [reply src/main/resources/eu/europa/ted/eforms/sdk/analysis/drools/xmlNoticeRules.drl]: Yes, a tab was used by mistake, but this was corrected in the recent commit that removed the "and" operator.

**Review by rouschr — APPROVED** [review]

**Pull request merged**